### ISSUE TITLE
Combobox keyboard navigation does not work when VoiceOver is active

### ISSUE THREAD

**Issue by berkerdemirer** [tailwindlabs/headlessui#3810]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

For example: v2.2.9

**What browser are you using?**

For example: Firefox

**Reproduction URL**

https://headlessui.com/react/combobox#installation 

You can use the example combobox from the docs.

**Describe your issue**

Activate VoiceOver, focus on the combobox and open the options. Try to navigate using arrow keys. It won't work.
